core/src/main/java/com/amplifyframework/logging/BroadcastLogger.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.logging;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.amplifyframework.util.Immutable;
+
+import java.time.LocalDateTime;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * An implementation of {@link Logger} that can store logs and
+ * uses a provided logger to emit logs.
+ */
+final class BroadcastLogger implements Logger {
+    // Maximum number of logs to store.
+    private static final int MAX_NUM_LOGS = 500;
```

Comment:
This is logic that you'll put in your `PersistentLogger`.

The `BroadcastLogger` just farms work out to an `AndroidLogger` (that emits to logcat) and a `PersistentLogger`, which buffers content for display in the dev menu.